lib/chef/knife/windows_listener_create.rb

```diff
@@ -0,0 +1,109 @@
+# Author:: Mukta Aphale (<mukta.aphale@clogeny.com>)
+# Copyright:: Copyright (c) 2014-2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../knife"
+require_relative "winrm_base"
+
+class Chef
+  class Knife
+    class WindowsListenerCreate < Knife
+      deps do
+        require "openssl"
+      end
+
+      banner "knife windows listener create (options)"
+
+      option :cert_install,
+        short: "-c CERT_PATH",
+        long: "--cert-install CERT_PATH",
+        description: "Adds specified certificate to the Windows Certificate Store's Local Machine personal store before creating listener."
+
+      option :port,
+        short: "-p PORT",
+        long: "--port PORT",
+        description: "Specify port. Default is 5986",
+        default: "5986"
+
+      option :hostname,
+        short: "-h HOSTNAME",
+        long: "--hostname HOSTNAME",
+        description: "Hostname on the listener. Default is blank",
+        default: ""
+
+      option :cert_thumbprint,
+        short: "-t THUMBPRINT",
+        long: "--cert-thumbprint THUMBPRINT",
+        description: "Thumbprint of the certificate. Required only if --cert-install option is not used."
+
+      option :cert_passphrase,
+        short: "-cp PASSWORD",
+        long: "--cert-passphrase PASSWORD",
+        description: "Password for certificate."
+
+      def get_cert_passphrase
+        print "Enter given certificate's passphrase (empty for no passphrase):"
+        passphrase = STDIN.gets
+        passphrase.strip
+      end
+
+      def run
+        STDOUT.sync = STDERR.sync = true
+
+        if Chef::Platform.windows?
+          begin
+            if config[:cert_install]
+              config[:cert_passphrase] = get_cert_passphrase unless config[:cert_passphrase]
+              result = `powershell.exe -Command " '#{config[:cert_passphrase]}' | certutil  -importPFX '#{config[:cert_install]}' AT_KEYEXCHANGE"`
```

Comment:
Same for this one. Should we do this via the win32-cert or simply refactor it to use a shellout